packages/core/src/generators/vue.ts

```diff
@@ -523,26 +531,27 @@ export const componentToVue =
             : ''
         }
         ${
-          component.hooks.onUpdate
-            ? !component.hooks.onUpdate.deps
-              ? // if we do not have dependencies, then we use `updated()` which re-runs on every render.
-                `updated() {
-                  ${processBinding(
-                    component.hooks.onUpdate.code,
-                    options,
-                    component,
-                  )}
-                },`
-              : // if we have dependencies, then we `watch` a computed property that combines the dependencies.
-                `watch: {
-                  ${ON_UPDATE_HOOK_NAME}() {
-                    ${processBinding(
-                      component.hooks.onUpdate.code,
-                      options,
-                      component,
-                    )}
+          onUpdateWithoutDeps.length
+            ? `updated() {
+            ${onUpdateWithoutDeps.map((hook) =>
+              processBinding(hook.code, options, component),
+            )}
```

Comment:
The `map` output automatically joins with `,`, which is giving us invalid code (see the `vue` test snapshots).

```suggestion
            ).join('\n')}
```